torchao/quantization/autoquant.py

```diff
@@ -433,20 +433,44 @@ def change_autoquantizable_to_quantized(model, **kwargs):
+    filter_fn=None, mode=["interpolate", .85], 
+    manual=False, 
+    **aq_kwargs
+):
     """
-    Wraps the given model in an AutoQuantWrapper. If `example_input` is provided, performs a forward pass on the input.
-    Otherwise, returns the wrapped model. The AutoQuantWrapper manages cases where the model is torch-compiled by first
-    performing autoquantization on the original model and then allowing the torch.compile run/tracing to occur.
+    Autoquantization is a process which identifies the fastest way to quantize each layer of a model over some set of potential
+    qtensor subclasses.
+    
+    Autoquantization happens in three steps:
+
+    1-Prepare Model: the model is searched for Linear layers whose weights are exchanged for AutoQuantizableLinearWeight.
+    2-Shape Calibration: the user runs the model on one or more inputs, the details of the activation shape/dtype seen by 
+        the AutoQuantizableLinearWeight are recorded so we know what shapes/dtypes to use in order to optimize the quantized op in step 3
+    3-Finalize Autoquantization: for each AutoQuantizableLinearWeight, benchmarks are run for each shape/dtype on each member of the qtensor_class_list.
+        the fastest option is picked, resulting in a highly performant model
+
+    This autoquant function performs step 1. Steps 2 and 3 can be completed by simply running the model.  
+    If `example_input` is provided, this function also runs the model (which completes steps 2 and 3). 
+    This autoquant api can handle models which have already had torch.compile applied to them, in which case, once the model is run and quantized, 
+    the torch.compile process normally proceeds as well.
+
+    To optimize over a combination of input shapes/dtypes, the user can set manual=True, run the model with all desired shapes/dtypes, then
+    call model.do_autoquant to finalize the quantization once the desired set of inputs have been logged.
 
     Args:
         model (torch.nn.Module): The model to be autoquantized.
         example_input (Any, optional): An example input for the model. If provided, the function performs a forward pass
-                                       on this input. Defaults to None.
+                                       on this input (which fully autoquantizes the model unless manual=True). Defaults to None.
         qtensor_class_list (list, optional): A list of tensor classes to be used for quantization. Defaults to DEFAULT_CLASS_LIST.
         filter_fn (callable, optional): A filter function to apply to the model parameters. Defaults to None.
         mode (list, optional): A list containing mode settings for quantization. The first element is the mode type (e.g., "interpolate"),
                                and the second element is the mode value (e.g., 0.85). Defaults to ["interpolate", .85].
+        manual (bool, optional): Whether to stop shape calibration and do autoquant after a single run (default, False) or to wait for 
```

Comment:
nice! much clearer

Reply:
How do you feel instead about calling this `defer_shape_calibration`?

Reply:
its not defering shape calibration, its elongating it

if anything it'd have to be defer_shape_calibration_end

Reply:
that is a mouthful lol, meh let's keep manual for now